### PR TITLE
Record time spent

### PR DESCRIPTION
**Pull request by dmcavoy**
# Background

#31429 created a UserLevelInfo table for recording further information about a UserLevel since the UserLevel table has grown very large and adding another column to it could cause site downtime.

UserLevelInfo has a column to record `time_spent` on a UserLevel. 

# Description

 This uses `timeSinceLastMilestone` which we track for milestone posts to keep track of the `time_spent` on an individual level by a particular user.

## Links

- [Record Time Spent Work Plan](https://docs.google.com/document/d/1sSxS1C0XHokLh34ObposdzI0I1FwCY7qJXmn0mtpwi0/edit?ts=5d8932a9&pli=1)
- [Progress Tab Spec](https://docs.google.com/document/d/11vid6ZWkpUV5TJm3swI6kbdSpg5_93UHDxnod58r_Zc/edit?pli=1#heading=h.on4pjo8b9e6i)
- [LP-916](https://codedotorg.atlassian.net/browse/LP-916)

## Testing story

Created new tests in `activities_controller_test.rb` to check that we create and update UserLevelInfo entries correctly. 

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by clareconstantine — APPROVED** [review]

**Review by davidsbailey — APPROVED** [review]
Looks good @dmcavoy !

File: dashboard/app/models/user_level_info.rb [hunk 1]
```diff
@@ -6,6 +6,10 @@
 #  time_spent    :integer          default(0)
 #  user_level_id :integer          unsigned
 #
+#
+# This model is meant to support the UserLevel model. The UserLevel table has gotten
+# too large and adding new columns to it can cause problems with production so we created
+# this model in order to store further information about a UserLevel.
```
Comment (Erin007) [inline dashboard/app/models/user_level_info.rb:13]: Thanks! This comment will help future folks who don't have the current context.

File: dashboard/app/controllers/activities_controller.rb [hunk 2]
```diff
@@ -213,6 +215,23 @@ def track_progress_for_user
...
+    if user_level
+      user_level_info = UserLevelInfo.find_by(
+        user_level_id: user_level.id
+      )
+    end
+
+    if user_level_info
+      # Add past time spent to current time spent on level
+      user_level_info.update(time_spent: [user_level_info.time_spent + [params[:timeSinceLastMilestone].to_i, 0].max, MAX_INT_MILESTONE].min)
+    else
```
Comment (Erin007) [inline dashboard/app/controllers/activities_controller.rb:230]: What's `MAX_INT_MILESTONE`? Something that already existed?

**Review by Erin007 — APPROVED** [review]

File: dashboard/app/controllers/activities_controller.rb (hunk 2 above)
Comment (dmcavoy) [reply dashboard/app/controllers/activities_controller.rb:230]: It already existed to make sure we didn't exceed the biggest number we could put in for time so I kept that same practice with timeSinceLastMilestone

**Pull request merged**